### PR TITLE
Add more SIMD spec tests

### PR DESCRIPTION
**Pull request by abrown**
These can be run once https://github.com/CraneStation/cranelift/commit/dde8ccf351cbbf7c5872c0ce6f8c3a2af2520a14 appears in a published crate.

### CODE REVIEW

**Comment by abrown** [comment]
This still needs to bump the `Cargo.toml`s to `0.45.0` but when I do that I see:

```
error[E0050]: method `define_function_body` has 3 parameters but the declaration in trait `cranelift_wasm::environ::spec::ModuleEnvironment::define_function_body` has 4
   --> wasmtime-environ/src/module_environ.rs:322:9
    |
322 | /         &mut self,
323 | |         body_bytes: &'data [u8],
324 | |         body_offset: usize,
    | |__________________________^ expected 4 parameters, found 3
```

Perhaps we should do that change in a separate PR.

**Comment by abrown** [comment]
@sunfishcode, can you review?

**Comment by sunfishcode** [comment]
@abrown On the CI, the Linux failure looks like a SIMD constant mismatch. Do you know what's causing that?

**Comment by sunfishcode** [comment]
The PR itself looks good to me, so the outstanding question here is what's behind the test failures in CI.

**Comment by abrown** [comment]
@sunfishcode I wasn't seeing those errors locally so I rebased and pushed to kick off another build. Doing this introduces changes from #434 but let's see what happens.

**Comment by abrown** [comment]
Hm, thinking about it some more, maybe it was that glibc floating point parsing issue; maybe one of the tests I added had the same problem and I was supposed to ignore it along with `simd_const`. In any case, I think there are new issues now that I rebased:
```
thread 'Cranelift::single_test_simd_align::simd_align' panicked at 'couldn't convert V128Const(I32x4([50462976, 117835012, 286263560, 353637138])) to a runtime value', wasmtime-wast/src/wast.rs:22:18
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/libunwind.rs:88
...
```
I wonder if this has something to do with #434; will look at it more next week.

**Comment by abrown** [comment]
I'm at a loss. I rebased to bring in the fix to #441 but Windows still fails with an unknown error: https://dev.azure.com/CraneStation/Wasmtime/_build/results?buildId=1638&view=logs&jobId=a3b6e9de-3482-53a8-3ded-9fe7718dae04&taskId=714523b9-99ae-5a52-140c-fd41a17c012d&lineStart=208&lineEnd=209&colStart=1&colEnd=1. The last thing I see in that PR is a bunch of ignored multi-value tests (which I think is currently correct?); @fitzgen, do you see anything obviously wrong with this PR?

File: build.rs [hunk 1]
```diff
@@ -167,11 +181,11 @@ fn write_testsuite_tests(
 
 /// Ignore tests that aren't supported yet.
 fn ignore(testsuite: &str, name: &str, strategy: &str) -> bool {
-    let is_multi_value = testsuite.ends_with("multi-value");
+    let is_multi_value = testsuite.ends_with("multi_value");
```
Comment (fitzgen) [inline build.rs]: In my submodule, the path is `spec_testsuite/proposals/multi-value` with a hyphen, not an underscore. Was this change and the similar one elsewhere intentional?

File: build.rs [hunk 2]
```diff
@@ -119,17 +132,17 @@ fn write_testsuite_tests(
...
         .to_str()
         .expect("to_str");
 
     writeln!(out, "        #[test]")?;
-    if ignore(testsuite, stemstr, strategy) {
+    if ignore(testsuite, testname, strategy) {
         writeln!(out, "        #[ignore]")?;
     }
-    writeln!(out, "        fn r#{}() {{", &stemstr.replace("-", "_"))?;
+    writeln!(out, "        fn r#{}() {{", &testname.replace("-", "_"))?;
```
Comment (fitzgen) [inline build.rs]: `testname` doesn't have the underscores, but the outputted test does, so...

File: build.rs [hunk 3]
```diff
@@ -119,17 +132,17 @@ fn write_testsuite_tests(
...
-    let stemstr = path
+    let testname = path
         .file_stem()
         .expect("file_stem")
         .to_str()
         .expect("to_str");
 
     writeln!(out, "        #[test]")?;
-    if ignore(testsuite, stemstr, strategy) {
+    if ignore(testsuite, testname, strategy) {
```
Comment (fitzgen) [inline build.rs]: ...therefore when we pass `testname` in to `ignore` here, it still has the hyphens, not the underscores.

Either we should replace all the hyphens early on, and then match `"multi_value"`, or we should replace the hyphens late and match `"multi-value"`.

Comment (abrown) [reply build.rs]: That's it! Thanks! I was trying to do the replacements early but at some point I either forgot about 136-139 or I rebased my changes away. Let's see if this latest push works.

**Comment by abrown** [comment]
Just to keep track of the weird things I'm observing for later reference. I see a segfault after many (all?) spec tests run on the Windows build check (which does `cargo test --release --all --exclude lightbeam --exclude wasmtime-wasi-c --exclude wasmtime-py --exclude wasmtime-api`): https://dev.azure.com/CraneStation/Wasmtime/_build/results?buildId=1662&view=logs&jobId=2f7369d1-fb97-5262-a571-f4e9c996b7d1&taskId=2ff2d7d3-48f5-51be-703f-27697ec3daa2&lineStart=152&lineEnd=153&colStart=1&colEnd=1.

**Comment by abrown** [comment]
Also for reference, I see the following rather unclear error even when running with `--nocapture` on:
```
running 89 tests
test Cranelift::misc_testsuite::misc_traps ... ignored
test Cranelift::misc_testsuite::stack_overflow ... ignored
test Cranelift::multi_value::binary ... ignored
test Cranelift::multi_value::block ... ignored
test Cranelift::multi_value::br ... ignored
test Cranelift::multi_value::call ... ignored
test Cranelift::multi_value::call_indirect ... ignored
test Cranelift::multi_value::fac ... ignored
test Cranelift::multi_value::func ... ignored
test Cranelift::multi_value::if ... ignored
test Cranelift::multi_value::loop ... ignored
test Cranelift::multi_value::type ... ignored
error: test failed, to rerun pass '-p wasmtime --test wast_testsuites'

##[error]Bash exited with code '127'.
```
It would seems as though the right things are being ignored but something else is going wrong. Perhaps a Windows path issue? https://dev.azure.com/CraneStation/Wasmtime/_build/results?buildId=1664&view=logs&jobId=a3b6e9de-3482-53a8-3ded-9fe7718dae04&taskId=714523b9-99ae-5a52-140c-fd41a17c012d&lineStart=193&lineEnd=208&colStart=1&colEnd=38

**Comment by peterhuene** [comment]
My guess would be that `skip-stack-guard-page` is running because it wasn't fixed to use underscores for the exclusion?  This is known to segfault because we're not restoring guard pages nor are we emitting unwind information for unwinding Wasm frames on the stack properly when we `longjmp`.

I have changes locally that enables all tests again on Windows, but it's dependent upon CraneStation/cranelift#1155.

**Comment by peterhuene** [comment]
Looking at the SIMD tests, they probably need to remain disabled on Windows as they trap and that isn't reliable yet without my changes.

**Comment by abrown** [comment]
@peterhuene, good to know; I was just trying to get things set up on Windows so I could debug what was going wrong but I will gladly disable them on Windows for now 😄.

**Comment by peterhuene** [comment]
I'll make sure to re-enable them with my changes (I'll consider it a requirement).

**Pull request merged**